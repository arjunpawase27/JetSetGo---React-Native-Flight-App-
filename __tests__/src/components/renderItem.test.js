import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import FlightRenderItem from '../../../src/components/renderItem';
import renderer from 'react-test-renderer';
describe('FlightRenderItem Component', () => {
  const item = {
    airline: "Air India",
    origin: "Bangalore",
    destination: "Kolkata",
    price: 6200,
    seatsAvailable: 110
  };
  describe('FlightRenderItem Component Snapshot', () => {
    const item = {
      airline: "Air India",
      origin: "Bangalore",
      destination: "Kolkata",
      price: 6200,
      seatsAvailable: 110
    };
  
    test('renders correctly', () => {
      const tree = renderer.create(<FlightRenderItem item={item} onPress={() => {}} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  test('renders correctly with given item data', () => {
    const { getByText } = render(
      <FlightRenderItem item={item} onPress={() => {}} />
    );

    expect(getByText(item.airline)).toBeDefined();
    expect(getByText(`${item.origin} -> ${item.destination}`)).toBeDefined();
    expect(getByText(`${item.price} â‚¹`)).toBeDefined();
    expect(getByText(`${item.seatsAvailable} Seats`)).toBeDefined();
  });

  test('calls onPress when the item is pressed', () => {
    const onPressMock = jest.fn();
    const { getByTestId } = render(
      <FlightRenderItem item={item} onPress={onPressMock} />
    );

    const touchableItem = getByTestId('flight-item-touchable');
    fireEvent.press(touchableItem);
    expect(onPressMock).toHaveBeenCalledWith(item);
  });
});
