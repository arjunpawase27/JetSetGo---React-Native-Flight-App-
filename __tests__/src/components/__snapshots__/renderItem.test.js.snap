// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightRenderItem Component FlightRenderItem Component Snapshot renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#fff",
      "borderRadius": 10,
      "elevation": 7,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginBottom": 10,
      "opacity": 1,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.5,
      "shadowRadius": 5.84,
    }
  }
  testID="flight-item-touchable"
>
  <View
    style={
      {
        "padding": 15,
      }
    }
  >
    <Text
      style={
        {
          "color": "#2f4149",
          "fontSize": 18,
          "fontWeight": "bold",
          "marginBottom": 5,
        }
      }
    >
      Air India
    </Text>
    <Text
      style={
        {
          "color": "#70a8c0",
          "fontSize": 16,
          "marginBottom": 5,
        }
      }
    >
      Bangalore -&gt; Kolkata
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#7bd3f7",
        "borderBottomRightRadius": 10,
        "borderTopRightRadius": 10,
        "justifyContent": "center",
        "paddingHorizontal": 15,
        "paddingVertical": 10,
        "width": 125,
      }
    }
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      6200
       â‚¹
    </Text>
    <Text
      style={
        {
          "color": "#70a8c0",
          "fontSize": 16,
        }
      }
    >
      110
       Seats
    </Text>
  </View>
</View>
`;
